### DISCUSSION TITLE
Is there any way to disable mDNS?

### DISCUSSION THREAD

**Discussion by wahidulislamriyad** [RavenSystem/esp-homekit-devices#1700] [Q&A]
Hello. I love RavenSystem's HAA! I am just installing this to all kind of smart devices in my home.
But I am having issue when using multiple HAA devices at the same time. When a HAA devices starts broadcasting mDNS, another HAA device gets unresponsive in Home app. That why, I was trying to move all HomeKit process in a single device and controlling other devices using HomeKit API. But those devices are also broadcasting mDNS.

In this situation, I see 2 solutions. Turn off mDNS broadcasting from other devices, or fix the firmware to support multiple HAA devices at the same time.

**Comment by RavenSystem** [comment]
mDNS is mandatory for HomeKit (and AirPrint, AirPlay...) and it can not be disabled. I have about 50-60 HAA devices connected to my network without any issue.

**Reply by wahidulislamriyad** [reply]
Thanks for your fast response.
Actually I wanted to make few of the devices to act as non-HomeKit device. As they have HomeKit API to control accessory state, that’s why I asked how to disable mDNS. But a main HAA device which will be the HomeBridge, will broadcast mDNS only.

I see that, when my iPhone tries to control any HomeKit device from Shortcut automation, many of the time, it fails. That’s because, mDNS stopped responding. And I have set TTL to 30 seconds for all devices.
For this reason, I wrote an SSH script which controls accessories via HomeKit API and Shortcut automation triggers this script when needed. That script is working on my router.

**Reply by RavenSystem** [reply]
Setting TTL to 30 for all devices will flood your network with broadcast traffic. Default value of 4500 seconds is better.

**Reply by wahidulislamriyad** [reply]
If I set TTL to 4500 seconds, most of the time HAA is unresponsive in Home app.
